sphinx/ext/intersphinx/_resolve.py

```diff
@@ -80,6 +80,10 @@ def _resolve_reference_in_domain_by_target(
             target_lower = target.lower()
             insensitive_matches = list(filter(lambda k: k.lower() == target_lower,
                                               inventory[objtype].keys()))
+            if len(insensitive_matches) > 1:
+                LOGGER.warning(__("%s: multiple matches found for %s:%s"),
+                               domain.name, objtype, target,
```

Comment:
```suggestion
                LOGGER.warning(__("multiple matches found for %s:%s"),
                               objtype, target,
```

the domain name is already part of the objtype, so it seems spurious to include it twice

Reply:
This reminds me approximately what I was thinking about when adding the `domain.name` to the log message: ideally it'd be useful (I think) to include the hyperlink to the inventory URL where the ambiguous entry originated.  I'll investigate that again within the next day or so.

Reply:
Hrm.  That doesn't seem so straightforward to me, because there isn't an `app` object (to retrieve the `config.intersphinx_mapping` from) in this context.

I'll commit the suggestion, it's an improvement.

Reply:
> > This reminds me approximately what I was thinking about when adding the `domain.name` to the log message: ideally it'd be useful (I think) to include the hyperlink to the inventory URL where the ambiguous entry originated. I'll investigate that again within the next day or so.
>
> Hrm. That doesn't seem so straightforward to me, because there isn't an `app` object (to retrieve the `config.intersphinx_mapping` from) in this context.

Perhaps this _is_ feasible, in fact.  Resolution is performed by the `IntersphinxRoleResolver` class, and that inherits a `self.app` attribute from `SphinxTransform` containing the Sphinx application that transformations are being applied to.  Let's find out whether it's possible to plumb that through to this function (`_resolve_reference_in_domain_by_target`).